kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaRequestHandler.java

```diff
@@ -2466,11 +2491,44 @@ private void updateProducerStats(final TopicPartition topicPartition, final int
             case ALTER:
             case UNKNOWN:
             case ALL:
-            case ANY:
             default:
-                return FutureUtil.failedFuture(
-                        new IllegalStateException("AclOperation [" + operation.name() + "] is not supported."));
+                break;
+        }
+        if (isAuthorizedFuture == null) {
+            return FutureUtil.failedFuture(
+                    new IllegalStateException("AclOperation [" + operation.name() + "] is not supported."));
         }
         return isAuthorizedFuture;
     }
+
+    /**
+     * If we are using kafkaEnableMultiTenantMetadata we need to ensure
+     * that the TenantSpec refer to an existing tenant.
+     * @param session
+     * @return whether the tenant is accessible
+     */
+    private boolean validateTenantAccessForSession(Session session)
+            throws AuthenticationException {
+        if (!kafkaConfig.isKafkaEnableMultiTenantMetadata()) {
+            // we are not leveraging kafkaEnableMultiTenantMetadata feature
+            // the client will access only system tenant
+            return true;
+        }
+        String tenantSpec = session.getPrincipal().getTenantSpec();
+        if (tenantSpec == null) {
+            // we are not leveraging kafkaEnableMultitenantMetadata feature
```

Comment:
Need Change to `kafkaEnableMultiTenantMetadata` too.

Reply:
sorry.
I have fixed this other occurrence. I did another pass and I can't find other points to change